dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -70,13 +81,65 @@ public IKernel Build()
     /// </summary>
     /// <param name="log">Logger to add.</param>
     /// <returns>Updated kernel builder including the logger.</returns>
+    [Obsolete("This method is deprecated and will be removed in one of the next SK SDK versions. Use AddLogging instead.")]
     public KernelBuilder WithLogger(ILogger log)
     {
         Verify.NotNull(log);
         this._logger = log;
         return this;
     }
 
+    /// <summary>
+    /// Add logging to the kernel to be built.
+    /// </summary>
+    /// <param name="logger">Logger to add.</param>
+    /// <returns>Updated kernel builder with enabled logging.</returns>
+    public KernelBuilder AddLogging(ILogger logger)
+    {
+        Verify.NotNull(logger);
+        this._logger = logger;
+        return this;
+    }
+
+    /// <summary>
+    /// Add metering to the kernel to be built.
+    /// </summary>
+    /// <param name="meterListener">Instance of <see cref="MeterListener"/> to be used for kernel metering.</param>
+    /// <returns>Kernel builder with enabled metering.</returns>
+    public KernelBuilder AddMetering(MeterListener meterListener)
+    {
+        Verify.NotNull(meterListener);
+
+        meterListener.InstrumentPublished = (instrument, listener) =>
```

Comment:
Great catch, I replaced assignment with `+=` to add callback method to invocation list.

Reply:
This behavior will allow to subscribe to all Kernel metering events. I added XML documentation which recommends using dedicated instance of `MeterListener` for `Kernel`. Since `MeterListener` is `IDisposable`, users will have control when to dispose and stop metering in Kernel. Same for `ActivityListener` and tracing.